cmd/kube-proxy/app/server.go

```diff
@@ -690,6 +692,19 @@ func (s *ProxyServer) Run() error {
 	// functions must configure their shared informer event handlers first.
 	informerFactory.Start(wait.NeverStop)
 
+	// Make an informer that selects for our nodename.
+	informerFactory = informers.NewSharedInformerFactoryWithOptions(s.Client, s.ConfigSyncPeriod,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("metadata.name", s.NodeRef.Name).String()
+		}))
+	nodeConfig := config.NewNodeConfig(informerFactory.Core().V1().Nodes(), s.ConfigSyncPeriod)
+	nodeConfig.RegisterEventHandler(s.Proxier)
+	go nodeConfig.Run(wait.NeverStop)
+
+	// This has to start after the calls to NewNodeConfig because that must
+	// configure the shared informer event handler first.
+	go informerFactory.Start(wait.NeverStop)
```

Comment:
I don't think you need to run this in a go routine, the informer factory runs each typed informer (in this case nodes only) in a separate go routine already.

Reply:
Yes, the code borrowed from tim and didn't notice that here need to change, gonna remove the `"go"`